integrations/oci/objectstorage/src/main/java/io/helidon/integrations/oci/objectstorage/OciObjectStorageHealthCheck.java

```diff
@@ -0,0 +1,233 @@
+
+/**
+ * Liveness check for an OCI's ObjectStorage bucket. Reads configuration to
+ * obtain bucket name as well as OCI properties from '~/.oci/config'.
+ */
+@Liveness
+@ApplicationScoped // this will be ignored if not within CDI
+@BuiltInHealthCheck
+public final class OciObjectStorageHealthCheck implements HealthCheck {
+    private static final Logger LOGGER = Logger.getLogger(OciObjectStorageHealthCheck.class.getName());
+
+    private static final int DEFAULT_TIMEOUT_SECONDS = 10;
+
+    private String bucket;
+    private String namespace;
+    private int timeout;
+    private OciObjectStorageRx ociObjectStorage;
+
+    private OciObjectStorageHealthCheck() {
+        this.timeout = DEFAULT_TIMEOUT_SECONDS;
+        Config ociConfig = Config.create().get("oci");
+        this.bucket = ociConfig.get("objectstorage.bucket").asString().orElse(null);
+        this.namespace = ociConfig.get("objectstorage.namespace").asString().get();
+        // This requires OCI configuration in ~/.oci/config
+        this.ociObjectStorage = OciObjectStorageRx.create(ociConfig);
+    }
+
+    private OciObjectStorageHealthCheck(Builder builder) {
+        this.timeout = builder.timeout;
+        this.bucket = builder.bucket;
+        this.namespace = builder.namespace;
+        this.ociObjectStorage = builder.ociObjectStorage;
+
+        Config ociConfig = builder.config;
+        if (ociConfig == null) {
+            ociConfig = Config.create().get("oci");
+        }
+        if (this.timeout == -1) {
+            this.timeout = ociConfig.get("objectstorage.healthcheck.timeout")
+                    .asInt().orElse(DEFAULT_TIMEOUT_SECONDS);
+        }
+        if (this.namespace == null) {
+            this.namespace = ociConfig.get("objectstorage.namespace").asString().get();
+        }
+        if (this.bucket == null) {
+            this.bucket = ociConfig.get("objectstorage.bucket").asString().orElse(null);
+        }
+        if (this.ociObjectStorage == null) {
+            // This requires OCI configuration in ~/.oci/config
+            this.ociObjectStorage = OciObjectStorageRx.create(ociConfig);
+        }
+    }
+
+
+    /**
+     * Internal validation method used for testing. Does not attempt to connect
+     * to OCI.
+     *
+     * @return outcome of internal validation.
+     */
+    boolean validate() {
+        try {
+            Objects.requireNonNull(bucket);
+            Objects.requireNonNull(namespace);
+            Objects.requireNonNull(ociObjectStorage);
+        } catch (NullPointerException e) {
+            return false;
+        }
+        return timeout > 0;
+    }
+
+    /**
+     * Create a new fluent API builder to configure a new health check.
+     *
+     * @return builder instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Create an instance.
+     *
+     * @return an instance.
+     */
+    public static OciObjectStorageHealthCheck create() {
```

Comment:
This method should not exist, as there are some required properties.
method `create(Config config)` would make sense here.